irobot_events_executor/src/rclcpp/timers_manager.cpp

```diff
@@ -24,8 +26,14 @@ TimersManager::~TimersManager()
   // Remove all timers
   this->clear();
 
-  // Make sure timers thread is stopped before destroying this object
-  this->stop();
+  // Make sure timers execution is stopped before destroying this object
+  this->notify_stop();
```

Comment:
Now that the thread is managed outside, do we still need this here?

Reply:
Yes I'm not sure this is needed, but as it was the same pattern as the `spin()` thread I added it